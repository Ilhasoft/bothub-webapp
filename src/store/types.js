export default {
  SET_TOKEN: 'SET_TOKEN',
  OPEN_LOGIN_MODAL: 'OPEN_LOGIN_MODAL',
  CLOSE_LOGIN_MODAL: 'CLOSE_LOGIN_MODAL',
  SET_PROFILE: 'SET_PROFILE',
  SET_NICKNAME_AUTHENTICATED: 'SET_NICKNAME_AUTHENTICATED',
  SET_REPOSITORY_RELATED_UUID: 'SET_REPOSITORY_RELATED_UUID',
  CACHE_FETCH: 'CACHE_FETCH',
};
