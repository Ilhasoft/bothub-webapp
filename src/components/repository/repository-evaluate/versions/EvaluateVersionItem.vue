<template>
  <div
    class="version-test-item">
    <router-link :to="resultRouterParams">
      <div
        class="version-test-item__wrapper">
        <div class="version-test-item__wrapper__info">
          <strong class="version-test-item__wrapper__info__item">
            {{ $t('webapp.result.test') }} #{{ version }}
          </strong>
          <language-badge
            :language="language"
            class="version-test-item__wrapper__info__item"
            main/>
          <small class="version-test-item__wrapper__info__item">
            {{ created_at | moment('from') }}</small>
        </div>
        <div class="version-test-item__wrapper__links">
          <router-link :to="resultRouterParams">
            <a> {{ $t('webapp.result.load_result') }} </a>
          </router-link>
        </div>
      </div>
    </router-link>
  </div>
</template>

<script>
import LanguageBadge from '@/components/shared/LanguageBadge';

export default {
  name: 'EvaluateVersionItem',
  components: {
    LanguageBadge,
  },
  props: {
    id: {
      type: Number,
      required: true,
    },
    version: {
      type: Number,
      required: true,
    },
    language: {
      type: String,
      required: true,
    },
    created_at: {
      type: String,
      default: '',
    },
  },
  computed: {
    resultRouterParams() {
      return {
        name: 'repository-result',
        params: {
          resultId: this.id,
        },
      };
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~@/assets/scss/utilities.scss';
@import '~@/assets/scss/colors.scss';
@import '~@/assets/scss/variables.scss';

.version-test-item {
  margin: 1rem 0;
  border: 1px solid $color-grey;
  border-radius: 0.5rem;
  font-family: $font-family;

  &__wrapper {
    display: flex;
    margin: .5rem;
    align-items: center;
    cursor: pointer;

    &__info {
      flex-grow: 1;
      align-self: center;

      &__item {
        margin: 0 .3rem;
      }
    }

    &__links {
      margin: .5rem;
    }
  }
}
</style>
