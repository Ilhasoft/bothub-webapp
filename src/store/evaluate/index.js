import actions from './actions';
import state from './state';
import mutations from './mutations';
import getters from './getters';
import types from './types';

export default {
  actions,
  state,
  mutations,
  getters,
  types,
};
