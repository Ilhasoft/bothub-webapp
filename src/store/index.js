import Vue from 'vue';
import Vuex from 'vuex';

import Auth from './auth';
import User from './user';
import Org from './org';
import Repository from './repository';
import Category from './category';
import Example from './example';
import EvaluateExample from './evaluate-example';
import Translate from './translate';
import CachedFetch from './cached-fetch';
import Entity from './entity';
import Intent from './intent';
import News from './news';
import Tutorial from './tutorial';
import External from './external';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Auth,
    User,
    Org,
    Repository,
    Category,
    Example,
    Translate,
    CachedFetch,
    EvaluateExample,
    Entity,
    News,
    Tutorial,
    External,
    Intent,
  },
});

store.dispatch('retriveAuthToken');

export default store;
