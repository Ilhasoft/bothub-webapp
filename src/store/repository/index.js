import state from './state';
import getters from './getters';
import actions from './actions';
import mutations from './mutations';
import types from './types';

export default {
  state,
  getters,
  actions,
  mutations,
  types,
};
