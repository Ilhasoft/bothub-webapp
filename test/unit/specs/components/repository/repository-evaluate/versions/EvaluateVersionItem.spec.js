import { shallowMount, createLocalVue } from '@vue/test-utils';
import EvaluateVersionItem from '@/components/repository/repository-evaluate/versions/EvaluateVersionItem';
import BH from 'bh';
import VueMoment from 'vue-moment';
import store from '@/store';


const localVue = createLocalVue();
localVue.use(BH);
localVue.use(VueMoment);

describe('EvaluateVersionItem.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(EvaluateVersionItem, {
      localVue,
      propsData: {
        id: 2,
        version: 2,
        language: 'pt',
        created_at: '2019-05-06T13:04:15.850503Z',
      },
      store,
    });
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
