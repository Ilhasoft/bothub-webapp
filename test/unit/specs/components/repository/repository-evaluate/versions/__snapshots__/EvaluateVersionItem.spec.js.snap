// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvaluateVersionItem.vue renders 1`] = `
<div class="version-test-item">
  <div class="version-test-item__wrapper">
    <div class="version-test-item__wrapper__check">
      <b-checkbox-stub truevalue="true"></b-checkbox-stub>
    </div>
    <div class="version-test-item__wrapper__info">
      <evaluate-type-tag-stub evaluatetype="Manual"></evaluate-type-tag-stub>
      <language-badge-stub language="pt" main="true" class="version-test-item__wrapper__info__item"></language-badge-stub> <span class="version-test-item__wrapper__info__item">
        some specific text #2
      </span> <small class="version-test-item__wrapper__info__item">
        in 9 months</small>
    </div>
    <div class="version-test-item__wrapper__links">
      <router-link to="[object Object]"><a> some specific text </a></router-link>/
      <router-link to="[object Object]"><a> some specific text </a></router-link>
    </div>
  </div>
</div>
`;
